comps/guardrails/pii_detection/pii_detection.py

```diff
@@ -117,19 +119,28 @@ def text_based_pii_detect(text_list: List[str], strategy, enable_ray=False, debu
         for data in tqdm(text_list, total=len(text_list)):
             if debug:
                 print("content is: ", data)
-            with Timer(f"detect pii on document {data[:50]} to Redis."):
+            with Timer(f"detect pii on document {data[:50]}"):
                 ret.append(pii_detector.detect_pii(data))
     return ret
 
 
 @register_microservice(
     name="opea_service@guardrails-pii-detection", endpoint="/v1/piidetect", host="0.0.0.0", port=6357
 )
-async def pii_detection(files: List[UploadFile] = File(None), link_list: str = Form(None), text_list: str = Form(None)):
+async def pii_detection(
+    files: List[UploadFile] = File(None),
+    link_list: str = Form(None),
+    text_list: str = Form(None),
+    strategy: str = Form(None),
+):
     if not files and not link_list and not text_list:
         raise HTTPException(status_code=400, detail="Either files, link_list, or text_list must be provided.")
 
-    strategy = "ner"  # Default strategy
+    if strategy is None:
+        strategy = "ner"
+
+    print("PII detection using strategy: ", strategy)
+
     pip_requirement = ["detect-secrets", "phonenumbers", "gibberish-detector"]
```

Comment:
This part should be a configurable setting?

Reply:
The strategy is configurable when sending request to the endpoint. The default is NER when the strategy is not specified in the http request.